server/channels/app/platform/service.go

```diff
@@ -323,11 +323,10 @@ func New(sc ServiceConfig, options ...Option) (*PlatformService, error) {
 	}
 
 	ps.AddLicenseListener(func(oldLicense, newLicense *model.License) {
-		if (oldLicense == nil && newLicense == nil) || !ps.startMetrics {
-			return
-		}
+		wasLicensed := (oldLicense != nil && *oldLicense.Features.Metrics) || (model.BuildNumber == "dev")
+		isLicensed := (newLicense != nil && *newLicense.Features.Metrics) || (model.BuildNumber == "dev")
 
-		if oldLicense != nil && newLicense != nil && *oldLicense.Features.Metrics == *newLicense.Features.Metrics {
+		if wasLicensed != isLicensed || !ps.startMetrics {
```

Comment:
I am slightly confused by this. We should be restarting the metrics if a new license gets added having metrics feature right? But in this case, it looks like any change from old to new means we don't do anything.

Reply:
You're right! I'm not sure how I missed this during testing. Let me explore the e2e suite to see if this is something I can automatically test.